### PR TITLE
Allow downloading LLVM project source for building `lld`

### PR DESCRIPTION
**Pull request by MaxDesiatov**
First step towards fixing https://github.com/apple/swift-sdk-generator/issues/9. To build `lld` on `x86_64` we need to download its sources first, and smallest source distributions are available at https://github.com/llvm/llvm-project/releases.

`DownloadableArtifacts` was also refactored to group hashes for different artifacts into fewer dictionaries.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test

File: Sources/SwiftSDKGenerator/Generator/LocalSwiftSDKGenerator.swift [hunk 1]
```diff
@@ -20,7 +20,7 @@ public final class LocalSwiftSDKGenerator: SwiftSDKGenerator {
   public let artifactID: String
   public let versionsConfiguration: VersionsConfiguration
   public let pathsConfiguration: PathsConfiguration
-  public let downloadableArtifacts: DownloadableArtifacts
+  public var downloadableArtifacts: DownloadableArtifacts
```
Comment (MaxDesiatov) [inline Sources/SwiftSDKGenerator/Generator/LocalSwiftSDKGenerator.swift:23]: Marked mutable as `useLLVMSources` is mutating.

File: Sources/SwiftSDKGenerator/Generator/LocalSwiftSDKGenerator.swift [hunk 2]
```diff
@@ -40,6 +40,7 @@ public final class LocalSwiftSDKGenerator: SwiftSDKGenerator {
       .removingLastComponent()
       .removingLastComponent()
       .removingLastComponent()
+      .removingLastComponent()
```
Comment (MaxDesiatov) [inline Sources/SwiftSDKGenerator/Generator/LocalSwiftSDKGenerator.swift:43]: Fix for `Bundles` ending up in the `Sources` directory after `LocalSwiftSDKGenerator.swift` was moved to a subdirectory.

**Comment by MaxDesiatov** [comment]
@swift-ci test

File: Sources/SwiftSDKGenerator/Artifacts/DownloadableArtifacts.swift [hunk 3]
```diff
@@ -13,71 +13,88 @@
...
+    case .macosx, .darwin:
+      self = .macOS
+    }
+  }
+
+  case linux(LinuxDistribution)
+  case macOS
+
+  /// No specific OS required for this artifact, it's not binary and is distributed in the source form.
+  case source
```
Comment (kateinoigakukun) [inline Sources/SwiftSDKGenerator/Artifacts/DownloadableArtifacts.swift:31]: Is this used anywhere? If it's not used now and we don't have a corresponding return value on this case for `llvmBinaryURLSuffix`, would it make sense to remove this case?

**Review by kateinoigakukun — APPROVED** [review]

File: Sources/SwiftSDKGenerator/Artifacts/DownloadableArtifacts.swift (hunk 3 above)
Comment (MaxDesiatov) [reply Sources/SwiftSDKGenerator/Artifacts/DownloadableArtifacts.swift]: Good catch, I've cleaned that up.

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Pull request merged**